astroid/rebuilder.py

```diff
@@ -1009,7 +1009,7 @@ def visit_comprehension(
         self, node: ast.comprehension, parent: NodeNG
     ) -> nodes.Comprehension:
         """Visit a Comprehension node by returning a fresh instance of it."""
-        newnode = nodes.Comprehension(parent)
+        newnode = nodes.Comprehension(parent=parent)
```

Comment:
Can you update this to also `provide` `lineno` etc? It's available on the `node`. No reason not to pass it.

Reply:
Apparently it doesn't? https://docs.python.org/3/library/ast.html#abstract-grammar

Reply:
Oh that's weird. I shouldn't have trusted my IDE which said it should 😓 

I'm wondering: what do you think of still providing `None` as values? That way we can at least remove the default values from `NodeNG` at some point.

Reply:
Sounds reasonable, although without a comment explaining why there are no line numbers I don't think it will be any clearer!

Reply:
Let's do that then!